cwltool/pathmapper.py

```diff
@@ -139,6 +144,30 @@ def trim_listing(obj):
     if obj.get("location", "").startswith("file://") and "listing" in obj:
         del obj["listing"]
 
+# Download http Files
+def downloadHttpFile(httpurl):
+    cache_session = None
+    if "HOME" in os.environ:
```

Comment:
Lets make this compliant with the [XDG Base Directory Specification](https://standards.freedesktop.org/basedir-spec/basedir-spec-0.6.html)'s `$XDG_CACHE_HOME` feature (if absent then the default is `$HOME/.config`) and if HOME isn't specified, then you can use [`os.path.expanduser('~')`](https://docs.python.org/3/library/os.path.html#os.path.expanduser) which means the bits with `TMP` and `/tmp` below aren't needed